adagrams/game.py

```diff
@@ -1,11 +1,116 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+    ("A", "E", "I", "O", "U", "L", "N", "R", "S", "T"): 1,
+    ("D", "G"): 2,
+    ("B", "C", "M", "P"): 3,
+    ("F", "H", "V", "W", "Y"): 4,
+    ("K", ): 5,
+    ("J", "X"): 8,
+    ("Q", "Z"): 10
+}
+
 def draw_letters():
-    pass
+    # Create a copy of LETTER_POOL so it won't change the original data
+    letter_pool_copy = LETTER_POOL.copy()   
+    drawn_letters = []                           
+    
+    for i in range(10):
+        # Choose a random letter 10 times
+        letter = random.choice(list(letter_pool_copy.keys()))
+
+        # Check letter qty, updating the LETTER_POOL_copy to reflect the distribution 
+        #   if qty == 1 remove it from the copy so it won't be chosen again
+        #   if qty > 1, decrease the qty by 1
+        if letter_pool_copy[letter] == 1:
+            del letter_pool_copy[letter]       
+        else:
+            letter_pool_copy[letter] -= 1
+        drawn_letters.append(letter)
+    return drawn_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # Create a copy of letter_bank so it won't change the original data
+    letter_bank_copy = letter_bank.copy()  
+    word = word.upper() 
+    
+    # Check if each single character exists in leeter_bank_copy:
+    #   if not, return False. 
+    #   if yes, remove it from letter_bank_copy
+    # If all letters in the letter_bank_copy, return True
+    for char in word:                               
+        if char not in letter_bank_copy:        
+            return False            
+        letter_bank_copy.remove(char)               
+    return True
 
 def score_word(word):
-    pass
+    score = 0
+    word = word.upper()
+    
+    # Check each character in the word to find it's score in SCORE_CHART:
+        # sum the score of each chacracter
+    for letter in word:                            
+        for char, value in SCORE_CHART.items():    
+            if letter in char:                     
+                score += value
```

Comment:
The dictionary structure of SCORE_CHART is not optimal for this problem.  The setup here with tuples as keys makes it necessary to search for the letter, but a dictionary where each letter is a key and the point value is the dictionary value would remove the need for this inner loop.